### PR TITLE
Add cargo.lock file

### PR DESCRIPTION
**Pull request by chenrui333**
relates to https://github.com/Homebrew/homebrew-core/issues/46025

It would be nice to have a cargo.lock file to enforce reproducible build.

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -72,12 +72,12 @@ jobs:
         uses: actions-rs/cargo@v1
         with:
           command: build
-          args: --verbose
+          args: --locked --verbose
```
Comment (lu-zero) [inline .github/workflows/rust.yml]: I'd rather have a separate set of CI targets for that.

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,2 @@
 /target
 **/*.rs.bk
-Cargo.lock
```
Comment (lu-zero) [inline .gitignore]: I'd rather have a CI action to add the lock to the release. @Luni-4 can you bake the same magic you put on rav1e please? :)

**Comment by lu-zero** [comment]
Please refer to https://github.com/lu-zero/cargo-c/issues/44#issuecomment-558784195

Having Cargo.lock in the git tree is highly disruptive

**Comment by Luni-4** [comment]
@chenrui333 

You could create a `deploy.yml` script that does the following steps:
- Run `cargo update` to generate the `Cargo.lock` file
- Update `Cargo.lock` alongside with the release, look at [here](https://github.com/xiph/rav1e/blob/master/.github/workflows/deploy.yml#L97)

The `deploy.yml` file runs when a new tag will be created. Look at [here](https://github.com/xiph/rav1e/blob/master/.github/workflows/deploy.yml#L3) to set tags for `GitHub Actions`. In this case we need only the `'v*.*.*'` pattern.

If you need some help, ping me :)

**Comment by anthraxx** [comment]
Is this issue not resolved with #66? That would be great to have reproducible builds enabled easily.

**Comment by lu-zero** [comment]
Depends on what you use. The `.crate` package provided by crates.io already contains a Cargo.lock (that's what I'm using in Gentoo). The next release will provide the same `Cargo.lock` as release item in case you cannot use the `.crate` package.

**Comment by Luni-4** [comment]
The `Cargo.lock` file is added to a new release, so we can close this one

**Pull request closed without merging**